components/organisms/Header/LanguageSelect/index.js

```diff
@@ -39,18 +28,8 @@ export default function LanguageSelect({ className }) {
     const newLocale = isDefaultLocale ? "es" : "en";
 
     if (newLocale !== locale) {
-      const parts = pathname?.split("/") || [];
-      parts.shift();
-
-      if (locale !== fallbackLng) {
-        parts.shift();
-      }
-
-      const route = `/${newLocale}/${parts.join("/")}${filterSearchParams(
-        searchParams
-      )}`;
       setLoading(true);
-      router.replace(route, { scroll: false });
+      router.replace({ pathname, query: searchParams }, { locale: newLocale });
```

Comment:
use `next-intl` router helper that will change the locale cookie on locale change